streams-bootstrap-core/src/testFixtures/java/com/bakdata/kafka/KafkaContainerHelper.java

```diff
@@ -0,0 +1,142 @@
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class
+        ));
+    }
+
+    public ReaderBuilder read() {
+        return new ReaderBuilder(Map.of(
+                ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class,
+                ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class
+        ));
+    }
+
+    public ImprovedAdminClient admin() {
+        return ImprovedAdminClient.create(Map.of(
+                AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG, KafkaContainerHelper.this.kafkaCluster.getBootstrapServers()
+        ));
+    }
+
+    @RequiredArgsConstructor
+    public class SenderBuilder {
+
+        private final @NonNull Map<String, Object> properties;
+
+        public SenderBuilder with(final String key, final Object value) {
+            final Map<String, Object> newProperties = new HashMap<>(this.properties);
+            newProperties.put(key, value);
+            return new SenderBuilder(Map.copyOf(newProperties));
+        }
+
+        public <K, V> void to(final String topic, final Iterable<? extends KeyValue<K, V>> records) {
+            final Map<String, Object> producerConfig = new HashMap<>(this.properties);
+            producerConfig.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG,
+                    KafkaContainerHelper.this.kafkaCluster.getBootstrapServers());
+            try (final Producer<K, V> producer = new KafkaProducer<>(producerConfig)) {
+                records.forEach(kv -> producer.send(new ProducerRecord<>(topic, kv.key, kv.value)));
+            }
+        }
+
+    }
+
+    @RequiredArgsConstructor
+    public class ReaderBuilder {
+
+        private final @NonNull Map<String, Object> properties;
+
+        public ReaderBuilder with(final String key, final Object value) {
+            final Map<String, Object> newProperties = new HashMap<>(this.properties);
+            newProperties.put(key, value);
+            return new ReaderBuilder(Map.copyOf(newProperties));
+        }
+
+        public <K, V> List<ConsumerRecord<K, V>> from(final String output, final Duration timeout) {
+            final Map<String, Object> consumerConfig = new HashMap<>(this.properties);
+            consumerConfig.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
+                    KafkaContainerHelper.this.kafkaCluster.getBootstrapServers());
+            try (final Consumer<K, V> consumer = new KafkaConsumer<>(consumerConfig)) {
+                final List<PartitionInfo> partitionInfos = consumer.listTopics().get(output);
+                final List<TopicPartition> topicPartitions = partitionInfos.stream()
+                        .map(partition -> new TopicPartition(partition.topic(), partition.partition()))
+                        .collect(Collectors.toList());
+                consumer.assign(topicPartitions);
+                consumer.seekToBeginning(topicPartitions);
```

Comment:
Calling `seekToBeginning` here is not what I would expect when using this method. I sometimes have tests that include multiple cycles of producing + consuming/validating.

Reply:
Without using seek to beginning, nothing is read if topic already has messages. Also this mimics the behavior of the read method from old embedded Kafka

Reply:
We could add `consumerConfig.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest");` instead

Reply:
I can do but is there any advantage over what we currently do? Also this is just test code

Reply:
Ah okay, I thought this is part of a library that can be user by others (me :smile:)

Reply:
Yeah, in a next step that would be great (also with support for multi-broker clusters)